metacat-connector-hive/src/test/groovy/com/netflix/metacat/connector/hive/HiveConnectorDatabaseSpec.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.connector.hive
+
+import com.netflix.metacat.common.QualifiedName
+import com.netflix.metacat.common.dto.Pageable
+import com.netflix.metacat.common.server.MetacatDataInfoProvider
+import com.netflix.metacat.common.server.connectors.ConnectorContext
+import com.netflix.metacat.common.server.connectors.model.DatabaseInfo
+import com.netflix.metacat.common.server.exception.DatabaseAlreadyExistsException
+import com.netflix.metacat.common.server.exception.DatabaseNotFoundException
+import com.netflix.metacat.connector.hive.converters.HiveConnectorInfoConverter
+import org.apache.hadoop.hive.metastore.api.Database
+import org.apache.thrift.TException
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+/**
+ * Unit test for HiveConnectorDatabaseSpec.
+ * @author zhenl
+ */
+class HiveConnectorDatabaseSpec extends Specification{
+    @Shared
+    MetacatHiveClient metacatHiveClient = Mock(MetacatHiveClient);
+    @Shared
+    HiveConnectorDatabaseService hiveConnectorDatabaseService = new HiveConnectorDatabaseService( this.metacatHiveClient, new HiveConnectorInfoConverter() )
+    @Shared
+    ConnectorContext connectorContext = new ConnectorContext(1, null);
+
+    def setupSpec() {
+        metacatHiveClient.getAllDatabases() >> ["test1", "test2", "dev1", "dev2"]
+        metacatHiveClient.getDatabase("testdb2") >> { throw new TException() }
+        metacatHiveClient.createDatabase(new Database("testdb3","testdb3","",Collections.EMPTY_MAP)) >> {}
+        metacatHiveClient.createDatabase(new Database("testdb2","testdb2","",Collections.EMPTY_MAP)) >> { throw new TException()}
+    }
+
+    @Unroll
```

Comment:
why is unroll on all of these when you're not actually running multiple tests using a table?